probe/probe.go

```diff
@@ -579,12 +627,84 @@ func getAnnotatedPodLabels(m map[string]string) mapset.Set[string] {
 	return b
 }
 
-func getAnnotatedPods(c *k8s.Client) error {
+func getNsSecurityPostureAndVisibility(c *k8s.Client, postureData map[string]string) (map[string]*NamespaceData, error) {
+	// Namespace/host security posture and visibility setting
+
+	mp := make(map[string]*NamespaceData)
+	namespaces, err := c.K8sClientset.CoreV1().Namespaces().List(context.Background(), metav1.ListOptions{})
+
+	if err != nil {
+		return mp, err
+	}
+	for _, ns := range namespaces.Items {
+
+		filePosture := postureData["filePosture"]
+		capabilityPosture := postureData["capabilitiesPosture"]
+		networkPosture := postureData["networkPosture"]
+
+		if len(ns.Annotations["kubearmor-file-posture"]) > 0 {
+			filePosture = ns.Annotations["kubearmor-file-posture"]
+		}
+
+		if len(ns.Annotations["kubearmor-capabilities-posture"]) > 0 {
+			capabilityPosture = ns.Annotations["kubearmor-capabilities-posture"]
+		}
+
+		if len(ns.Annotations["kubearmor-network-posture"]) > 0 {
+			networkPosture = ns.Annotations["kubearmor-network-posture"]
+		}
+
+		mp[ns.Name] = &NamespaceData{
+			NsDefaultPosture:   tp.DefaultPosture{FileAction: filePosture, CapabilitiesAction: capabilityPosture, NetworkAction: networkPosture},
+			NsVisibilityString: ns.Annotations["kubearmor-visibility"],
+			NsVisibility: Visibility{
+				Process:      strings.Contains(ns.Annotations["kubearmor-visibility"], "process"),
+				File:         strings.Contains(ns.Annotations["kubearmor-visibility"], "file"),
+				Network:      strings.Contains(ns.Annotations["kubearmor-visibility"], "network"),
+				Capabilities: strings.Contains(ns.Annotations["kubearmor-visibility"], "capabilities"),
+			},
+			NsPostureString: "File(" + filePosture + "), Capabilities(" + capabilityPosture + "), Network (" + networkPosture + ")",
+		}
+	}
+	return mp, err
+}
+
+// NamespaceData structure definition
+type NamespaceData struct {
```

Comment:
Let's create a new types.go in probe package and move all the probe type definitions to there for better readability.